applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c

```diff
@@ -190,37 +199,49 @@ K_WORK_DEFINE(cap_start_work, cap_start_worker);
  * @retval	0	Operation successful.
  * @retval	-EINVAL	Any error.
  */
-static int device_pres_delay_find(uint8_t index, uint32_t *pres_dly_us)
+static int device_pres_delay_find(struct stream_index stream_idx, uint32_t *pres_dly_us)
 {
-	uint32_t pd_min = unicast_servers[index].sink_ep->qos_pref.pd_min;
-	uint32_t pd_max = unicast_servers[index].sink_ep->qos_pref.pd_max;
-	uint32_t pref_pd_min = unicast_servers[index].sink_ep->qos_pref.pref_pd_min;
-	uint32_t pref_pd_max = unicast_servers[index].sink_ep->qos_pref.pref_pd_max;
-
-	LOG_DBG("Index: %d, Pref min: %d, pref max: %d, pres_min: %d, pres_max: %d", index,
+	uint8_t lvl1_idx = stream_idx.lvl1_idx;
+	uint8_t lvl2_idx = stream_idx.lvl2_idx;
+	uint8_t lvl3_idx = stream_idx.lvl3_idx;
+
+	uint32_t pd_min = unicast_servers[lvl1_idx][lvl2_idx][lvl3_idx].sink_ep->qos_pref.pd_min;
+	uint32_t pd_max = unicast_servers[lvl1_idx][lvl2_idx][lvl3_idx].sink_ep->qos_pref.pd_max;
+	uint32_t pref_pd_min =
+		unicast_servers[lvl1_idx][lvl2_idx][lvl3_idx].sink_ep->qos_pref.pref_pd_min;
+	uint32_t pref_pd_max =
+		unicast_servers[lvl1_idx][lvl2_idx][lvl3_idx].sink_ep->qos_pref.pref_pd_max;
+
+	LOG_DBG("Index: %d, Pref min: %d, pref max: %d, pres_min: %d, pres_max: %d", lvl3_idx,
 		pref_pd_min, pref_pd_max, pd_min, pd_max);
 
 	*pres_dly_us = 0;
 
-	for (int i = 0; i < ARRAY_SIZE(unicast_servers); i++) {
-		if (le_audio_ep_qos_configured(unicast_servers[i].sink_ep)) {
+	for (int i = 0; i < ARRAY_SIZE(unicast_servers[lvl1_idx][lvl2_idx]); i++) {
+		if (le_audio_ep_qos_configured(unicast_servers[lvl1_idx][lvl2_idx][i].sink_ep)) {
 			LOG_DBG("i: %d, Pref min: %d, pref max: %d, pres_min: %d, pres_max: %d", i,
```

Comment:
Suggest we print the levels here as well.

Reply:
We do print `i` which is lvl3, lvl1 and lvl2 will always be 0 for unicasts (unless we start to support multiple CIGs)